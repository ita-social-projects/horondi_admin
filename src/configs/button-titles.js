const buttonTitles = {
  DELETE_TITLE: 'Видалити',
  EDIT_TITLE: 'Редагувати',
  CREATE_NEWS_TITLE: 'Додати новину',
  CREATE_PATTERN_TITLE: 'Додати гобелен',
  CREATE_MODEL_TITLE: 'Додати модель',
  MODEL_SAVE_TITLE: 'Зберегти',
  MODEL_CONSTRUCTOR: 'Конструктор',
  CANCEL_TITLE: 'Ні',
  LOGOUT_TITLE: 'Вихід',
  CREATE_BUSINESS_PAGE: 'Додати бізнес сторінку',
  CREATE_CONTACT_TITLE: 'Додати контакти',
  SWITCH_USER_STATUS_TITLE: 'Змінити статус користувача',
  USER_INACTIVE_TITLE: 'Деактивувати',
  USER_ACTIVE_TITLE: 'Активувати',
  ADD_CATEGORY: 'Додати категорію',
  GO_BACK_TITLE: 'Назад',
  ADD_CATEGORY_IMAGE: 'Додати зображення',
  ADD_CATEGORY_NAME: 'Додати назву',
  CANCEL: 'Відмінити',
  SAVE_CATEGORY: 'Зберегти категорію',
  CREATE_SPECIAL_USER: 'Створити спецкористувача',
  ADD_PHOTO_LABEL: '+',
  CREATE_CATEGORY: 'Створити категорію',
  PATTERN_REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цей гобелен?',
  REMOVE_CONTACT_MESSAGE: 'Ви впевнені, що хочете видалити цей контакт?',
  USER_UNACTIVE_TITLE: 'Деактивувати',
  SHOW_COMMENTS_TITLE: 'Переглянути коментарі',
  HIDE_COMMENTS_TITLE: 'Приховати коментарі',
  SAVE_ORDER: 'Зберегти замовлення',
  SAVE_TITLE: 'Зберегти',
  UNDERSTAND: 'Зрозуміло',
  ANSWER: 'Відповісти',
  TO_SPAM: 'У СПАМ',
  MOVE_ALL_TO_SPAM: 'Перемістити у СПАМ',
  CREATE_SIZE_TITLE: 'Створити розмір',
  SAVE_SIZE_TITLE: 'Зберегти розмір',
  CREATE_MATERIAL_TITLE: 'Створити матеріал',
  CREATE_COLOR_TITLE: 'Створити колір',
  UPDATE_COLOR_TITLE: 'Зберегти зміни',
  SAVE_MATERIAL: 'Зберегти матеріал',
  CREATE_HEADER_TITLE: 'Створити посилання',
  HEADER_SAVE_TITLE: 'Зберегти посилання',
  GO_TO_MATERIAL_COLOR_PALLET: 'Редагувати палітру',
  CLOSE_DIALOG_TITLE: 'Закрити вікно',
  ORDER_DETAILS: 'Деталі',
  USER_STATUS_TITLE: 'Статус',
  CLEAR_FILTERS: 'Очистити фільтри',
  CREATE_SLIDE_TITLE: 'Створити слайд',
  OPEN_SLIDE_EDIT: 'Увімкнути редагування',
  SAVE_SLIDE_ORDER: 'Зберегти зміни',
  CANCEL_SLIDE_ORDER: 'Скасувати зміни',
  AUTHOR_PHOTO: 'Фото Автора',
  MAIN_PHOTO: 'Головне Фото',
  CREATE_CONSTRUCTOR_BASIC_TITLE: 'Додати основу',
  CREATE_CONSTRUCTOR_FRONT_POCKET_TITLE: 'Додати передню кишеню',
  CREATE_CONSTRUCTOR_BOTTOM_TITLE: 'Додати низ',
  GO: 'Перейти',
  CATEGORY_TITLE: 'Категорія',
  LOOK: 'Переглянути',
  ADD_ORDER: 'Додати замовлення',
  USER_ROLE_TITLE: 'Роль',
  EVENT_TITLE: 'Подія',
  SEND_CONFIRM: 'Надіслати підтвердження',
  SEND_CODE: 'Надіслати код для підтвердження',
  EXIT_WITHOUT_SAVING: 'Вийти без збереження',
  ORDER_STATUS: 'Статус замовлення',
  PAYMENT_STATUS: 'Статус оплати',
  SIZE_NAME: 'Розмір',
  AVAILABLE: 'Доступність',
  PURPOSE: 'Застосування',
  CATEGORY: 'Категорії',
  PATTERN: 'Гобелени',
  MODEL: 'Моделі',
  YES_BUTTON: 'Так',
  NO_BUTTON: 'Ні',
  CREATE_BACK_TITLE: 'Додати спинку',
  CREATE_POCKETS_TITLE: 'Додати кишеню',
  PATTERN_AVAILABLE: 'Доступний',
  PATTERN_MATERIAL: 'Матеріал',
  CREATE_CONSTRUCTOR: 'Створити новий '
};

export default buttonTitles;
