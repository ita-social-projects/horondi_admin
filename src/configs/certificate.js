const certificate = {
  TITLE_CERTIFICATE: 'HORONDI',
  CERTIFICATE_TEXT_WITH_VALUE: (value) =>
    `Подарунковий сертифікат на ${value} грн`
};

export default certificate;
