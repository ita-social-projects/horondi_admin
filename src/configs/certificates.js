const certificates = {
  certificateStatus: {
    ACTIVE: 'Активний',
    USED: 'Використаний',
    EXPIRED: 'Протермінований'
  }
};

export default certificates;
