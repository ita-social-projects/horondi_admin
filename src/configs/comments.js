const comments = {
  commentType: {
    COMMENT: 'Коментар',
    REPLY_COMMENT: 'Відповідь'
  }
};

export default comments;
