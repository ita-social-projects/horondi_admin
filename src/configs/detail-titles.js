const detailTitles = {
  users: {
    avatar: { id: 'avatar' },
    name: { id: 'name' },
    status: { id: 'status' },
    sections: [
      { id: 'phone', label: 'Телефон' },
      { id: 'country', label: 'Країна' },
      { id: 'city', label: 'Місто' },
      { id: 'adress', label: 'Адреса' },
      { id: 'postCode', label: 'Поштовий індекс' }
    ]
  },
  emailQuestions: {
    customer: {
      date: 'Дата запитання:',
      status: 'Статус:',
      sender: 'Відправник:',
      email: 'Email:',
      question: 'Запитання:'
    },
    admin: {
      date: 'Дата відповіді:',
      admin: 'Адмін:',
      answer: 'Відповідь:'
    }
  }
};

export default detailTitles;
