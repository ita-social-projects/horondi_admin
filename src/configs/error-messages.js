export const errorMessages = {
  USER_NOT_FOUND: 'Користувач не знайдений!',
  INVALID_PERMISSIONS: 'Недостатньо прав користувача',
  WRONG_CREDENTIALS: 'Неправильно вказані вхідні дані',
  INPUT_NOT_VALID: 'Неправильні ввідні дані',
  USER_ALREADY_EXIST: 'Користувач з такими даними вже існує',
  INVALID_ADMIN_INVITATIONAL_TOKEN:
    'Неправильне посилання на створення користувача',
  NO_COLORS: 'Створіть колір',
  USER_IS_ALREADY_BLOCKED: 'Цей користувач вже неактивний',
  YOU_CANT_BLOCK_YOURSELF: 'Ви не можете деактивувати власний профіль',
  ONLY_SUPER_ADMIN_CAN_BLOCK_ADMIN: 'У вас недостатньо прав',
  USER_IS_ALREADY_UNLOCKED: 'Цей користувач вже активний',
  YOU_CANT_UNLOCK_YOURSELF: 'Ви не можете активувати власний профіль',
  ONLY_SUPER_ADMIN_CAN_UNLOCK_ADMIN: 'У вас недостатньо прав',
  INVALID_OTP_CODE: 'Неправильний otp-код'
};

export const loginErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PASSWORD_MIN_LENGTH_MESSAGE: 'Пароль повинен містити не менше 8 символів',
  PASSWORD_MAX_LENGTH_MESSAGE: 'Пароль повинен містити не більше 20 символів',
  PASSWORD_LANG_MESSAGE: 'Використовуйте латиницю різних регістрів та цифри',
  ENTER_FIRSTNAME_MESSAGE: "Введіть ім'я",
  ENTER_LASTNAME_MESSAGE: 'Введіть прізвище',
  ENTER_PASSWORD_MESSAGE: 'Введіть пароль',
  FIRSTNAME_MIN_LENGTH_MESSAGE: "Ім'я повинно містити не менше 2 символів",
  LASTNAME_MIN_LENGTH_MESSAGE: 'Прізвище повинно містити не менше 2 символів',
  FIRSTNAME_MAX_LENGTH_MESSAGE: "Ім'я повинно містити не більше 30 символів",
  LASTNAME_MAX_LENGTH_MESSAGE: 'Прізвище повинно містити не більше 30 символів',
  SELECT_ROLE_MESSAGE: 'Оберіть роль',
  WRONG_FORMAT: 'Не допускається використання спеціальних символів',
  ENTER_CODE: 'Введіть код'
};

export const commonErrorMessages = {
  MIN_LENGTH_MESSAGE: 'Мінімум 2 символи',
  MAX_LENGTH_MESSAGE: 'Максимум 50 символiв',
  MAX_LENGTH_MESSAGE_300: 'Не більше 300 символів',
  ERROR_MESSAGE: 'Поле не може бути порожнім',
  UA_NAME_MESSAGE: 'Поле може містити тільки українські літери та цифри',
  EN_NAME_MESSAGE: 'Поле може містити тільки англійські літери та цифри',
  ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри',
  PRICE_ERROR: 'Це поле повинно містити тільки додатні значенння'
};
export const promoCodeErrorMessages = {
  LENGTH_CODE: 'Поле має містити від 2 до 30 символів',
  STYLE_CODE: 'Поле має містити англійські літери або цифри',
  ERROR_MESSAGE: 'Поле має бути заповнене',
  LENGTH_DISCOUNT: 'Поле має містити максимально 2 цифри',
  POSITIVE_DISCOUNT: 'Тільки більше нуля',
  MULTIPLE_DISCOUNT: 'Число має бути кратним п’яти'
};

export const patternErrorMessages = {
  PATTERN_VALIDATION_ERROR_NAME: 'Назва повинна містити від 2 до 50 символів',
  PATTERN_VALIDATION_ERROR_DESCRIPTION:
    'Опис повинний містити від 2 до 1000 символів',
  PHOTO_NOT_PROVIDED: "Фото гобелена є обов'язковим",
  CONSTRUCTOR_PHOTO_NOT_PROVIDED: "Фото для конструктора є обов'язковим"
};

export const backErrorMessages = {
  PHOTO_NOT_PROVIDED: "Фото спинки є обов'язковим",
  BACK_UA_NAME_MESSAGE: 'Введіть назву спинки українською',
  BACK_EN_NAME_MESSAGE: 'Введіть назву спинки англійською'
};

export const bottomErrorMessages = {
  PHOTO_NOT_PROVIDED: "Фото низу є обов'язковим",
  BOTTOM_UA_NAME_MESSAGE: 'Введіть назву низу українською',
  BOTTOM_EN_NAME_MESSAGE: 'Введіть назву низу англійською'
};

export const closuresErrorMessages = {
  PHOTO_NOT_PROVIDED: "Фото защіпки є обов'язковим",
  CLOSURES_UA_NAME_MESSAGE: 'Введіть назву защіпки українською',
  CLOSURES_EN_NAME_MESSAGE: 'Введіть назву защіпки англійською'
};

export const pocketsErrorMessages = {
  POCKETS_POSITION_ERROR_MESSAGE: 'Оберіть позицію із запропронованого списку',
  POCKETS_ERROR: 'Додайте фото для кишені',
  POCKETS_UA_NAME_MESSAGE: 'Введіть назву кишені українською',
  POCKETS_EN_NAME_MESSAGE: 'Введіть назву кишені англійською'
};

export const basicsErrorMessages = {
  PHOTO_NOT_PROVIDED: "Фото основи є обов'язковим"
};

export const constructorErrorMessages = {
  PHOTO_NOT_PROVIDED: 'Додайте фото для конструктора'
};

export const strapsErrorMessages = {
  STRAPS_COLOR_ERROR_MESSAGE: 'Оберіть колір із запропонованого списку'
};

export const categoryErrorMessages = {
  CATEGORY_VALIDATION_ERROR: 'Це поле повинно містити від 2 до 30 символів',
  CATEGORY_VALIDATION_ERROR_CATEGORY_NAME:
    'Назва повинна містити від 2 до 50 символів',
  CATEGORY_CODE_MESSAGE: 'Поле не може містити спеціальні символи',
  CATEGORY_ERROR: 'Додайте фото для категорії'
};

export const headerErrorMessages = {
  HEADER_VALIDATION_ERROR: 'Введіть посилання для сторінки',
  NOT_UA_NAME_MESSAGE: `Введіть ім'я посилання українською`,
  NOT_EN_NAME_MESSAGE: `Введіть ім'я посилання англійською`
};

export const newsErrorMessages = {
  NAME_MIN_LENGTH_MESSAGE: `Ім'я автора повинне містити не менше 2 символів`,
  TITLE_MIN_LENGTH_MESSAGE: 'Заголовок повинен містити не менше 10 символів',
  TEXT_MIN_LENGTH_MESSAGE: 'Текстове поле повино містити не менше 10 символів',
  NOT_EN_TEXT_MESSAGE: 'Введіть текст новини англійською',
  NOT_UA_TEXT_MESSAGE: 'Введіть текст новини українською',
  NOT_EN_TITLE_MESSAGE: 'Введіть заголовок новини англійською',
  NOT_UA_TITLE_MESSAGE: 'Введіть заголовок новини українською',
  NOT_EN_AUTHOR_NAME_MESSAGE: "Введіть ім'я автора новини англійською",
  NOT_UA_AUTHOR_NAME_MESSAGE: "Введіть ім'я автора новини українською"
};

export const modelErrorMessages = {
  PHOTO_NOT_PROVIDED: 'Додайте фото для моделі',
  NOT_EN_DESCRIPTION_MESSAGE: `Введіть опис моделі англійською`,
  NOT_UA_DESCRIPTION_MESSAGE: `Введіть опис моделі українською`,
  NO_STRING_TYPE_MESSAGE: `Повинно містити букви, цифри та спецсимволи`
};

export const contactErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PHONE_NUMBER_LENGTH_MESSAGE:
    'Довжина номеру телефону повинна містити 12 символів',
  PHONE_NUMBER_TYPE_MESSAGE: 'Номер повинен містити лише числа',
  ENTER_PHONE_NUMBER_MESSAGE: 'Введіть номер',
  INPUT_LENGTH_MESSAGE: 'Довжина повинна містити не менше 10 символів',
  ENTER_UA_SCHEDULE_MESSAGE: 'Введіть розклад українською',
  ENTER_EN_SCHEDULE_MESSAGE: 'Введіть розклад англійською',
  ENTER_UA_ADDRESS_MESSAGE: 'Введіть адресу українською',
  ENTER_EN_ADDRESS_MESSAGE: 'Введіть адресу англійською',
  IMAGE_FORMAT_MESSAGE:
    'Введіть коректний формат, наприклад: https://example.com/',
  ENTER_LINK_MESSAGE: 'Введіть посилання',
  CONTACT_ERROR_MESSAGE: 'Помилка збереження',
  INVALID_PHONE_MESSAGE:
    'Введіть коректний формат, наприклад: +380(XX)-XXX-XX-XX'
};

export const emailQuestionsErrorMessages = {
  ANSWER_INPUT_MESSAGE: 'Введіть текст для відповіді'
};

export const materialErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 100 символів`,
  NOT_EN_DESCRIPTION_MESSAGE: `Введіть опис матеріалу англійською`,
  NOT_UA_DESCRIPTION_MESSAGE: `Введіть опис матеріалу українською`
};

export const sizeErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 20 символів`,
  MIN_LENGTH_MESSAGE: `Не менше 1см`,
  MAX_LENGTH_MESSAGE_SIZE: `Не більше 35см`,
  MIN_LENGTH_MESSAGE_SIZE: `Не менше 1 символу`,
  MAX_WEIGHT_MESSAGE_SIZE: `Не більше 5 кг`,
  MIN_WEIGHT_MESSAGE_SIZE: `Не менше 0.1 кг`,
  NO_NUMBER_TYPE_MESSAGE: `Повинно містити число`,
  NO_STRING_TYPE_MESSAGE: `Повинно містити букви`
};

export const colorErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 100 символів`,
  COLOR_VALIDATION_ERROR: `Неправильний формат кольору`,
  NOT_EN_SIMPLE_NAME_MESSAGE: `Введіть просту назву кольору англійською`,
  NOT_UA_SIMPLE_NAME_MESSAGE: `Введіть просту назву кольору українською`
};

export const statsErrorMessages = {
  NO_STATS: 'Статистика для вибраного значення відсутня'
};

export const homePageSlideErrorMessages = {
  NOT_UA_NAME_MESSAGE: `Введіть ім'я слайду українською`,
  NOT_EN_NAME_MESSAGE: `Введіть ім'я слайду англійською`,
  NOT_EN_DESCRIPTION_MESSAGE: `Введіть опис слайду англійською`,
  NOT_UA_DESCRIPTION_MESSAGE: `Введіть опис слайду українською`
};

export const paginationInputErrorMessages = {
  MUST_BE_NUMBER: 'Тільки цифри',
  MUST_BE_POSITIVE: 'Тільки більше нуля',
  PAGE_NOT_FOUND: 'Сторінку не знайдено'
};

export const businessPageErrorMessages = {
  ENTER_CODE_ERROR_MESSAGE: 'Введіть унікальний ідентифікатор для сторінки',
  ENTER_TITLE_ERROR_MESSAGE: 'Введіть заголовок',
  ENTER_TEXT_ERROR_MESSAGE: 'Введіть текст для сторінки',
  MIN_TEXT_LENGTH_MESSAGE: 'Мінімум 10 символiв'
};

export const productErrorMessages = {
  REQUIRED_PHOTOS: "Головне фото для продукту є обов'язковим",
  REQUIRED_FIELD: "Обов'язкове поле",
  NAME_TOO_SHORT_MESSAGE: 'Назва продукту повинна містити не менше 6 символів',
  NAME_TOO_LONG_MESSAGE: 'Назва продукту повинна містити не більше 50 символів',
  CORRECT_DATA_ERROR:
    'Заповніть необхідні поля коректними даними для усіх мов ',
  NOT_UA_NAME_MESSAGE: 'Введіть назву українською',
  NOT_EN_NAME_MESSAGE: 'Введіть назву англійською'
};

export const userErrorMessages = {
  USER_FIRSTNAME_MESSAGE: "Введіть коректне ім'я",
  USER_FIRSTNAME_MIN_LENGTH_MESSAGE: "Ім'я повинно містити не менше 2 символів",
  USER_FIRSTNAME_MAX_LENGTH_MESSAGE:
    "Ім'я повинно містити не більше 30 символів",
  USER_LASTNAME_MESSAGE: 'Введіть коректне прізвище',
  USER_LASTNAME_MIN_LENGTH_MESSAGE:
    'Прізвище повинно містити не менше 2 символів',
  USER_LASTNAME_MAX_LENGTH_MESSAGE:
    'Прізвище повинно містити не більше 30 символів',
  USER_PASSWORD_MESSAGE: 'Введіть пароль',
  USER_PASSWORD_MIN_LENGTH_MESSAGE:
    'Пароль повинен містити не менше 8 символів',
  USER_PASSWORD_MAX_LENGTH_MESSAGE:
    'Пароль повинен містити не більше 20 символів',
  USER_PASSWORD_LANG_MESSAGE:
    'Використовуйте латиницю різних регістрів та цифри',
  USER_EMAIL_MESSAGE: 'Введіть email',
  USER_INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  USER_PHONE_NUMBER_MESSAGE: 'Введіть номер телефону',
  USER_INVALID_PHONE_NUMBER_MESSAGE: 'Введіть номер телефону',
  USER_INVALID_ADDRESS_MESSAGE: 'Введіть коректну адресу',
  USER_ERROR_MESSAGE: 'Поле не може бути порожнім'
};
