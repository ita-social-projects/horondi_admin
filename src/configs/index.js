import { colors } from '@material-ui/core';
import routes from './routes';
import {
  menuCategories,
  clientMenuCategories,
  catalogMenuCategories,
  staticPagesCategories,
  constructorMenuCategories
} from './menu-categories';
import tableHeadRowTitles from './table-head-row-titles';
import detailTitles from './detail-titles';
import buttonTitles from './button-titles';
import statuses from './statuses';
import filterLabels from './filter-labels';
import materialUiConstants from './material-ui-constants';
import {
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  patternErrorMessages,
  backErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  errorMessages,
  categoryErrorMessages,
  modelErrorMessages,
  headerErrorMessages,
  homePageSlideErrorMessages,
  constructorErrorMessages,
  statsErrorMessages,
  paginationInputErrorMessages,
  commentErrorMessages,
  replyCommentErrorMessages,
  sizeErrorMessages,
  businessPageErrorMessages,
  pocketsErrorMessages
} from './error-messages';
import messages from './messages';
import formRegExp from './form-regexp';
import labels from './labels';
import { newsPaginationPayload, contactsPaginationPayload } from './pagination';
import titles from './titles';
import languages from './languages';
import {
  imageUploadNewsInputsId,
  valueKeys,
  inputNames
} from './news-form-helpers';

export const config = {
  fetchPolicy: 'no-cache',
  app: {
    title: 'Horondi Admin Portal',
    serverUrl: 'http://localhost:5000/',
    drawerWidth: 230,
    snackBarDuration: 4000,
    rowsPerPageOptions: [10, 25, 50, 100]
  },
  languages,
  tokenName: 'HORONDI_AUTH_TOKEN',
  templates: {
    categoryTemplate: {
      available: false,
      code: '',
      images: {
        large: '',
        medium: '',
        small: '',
        thumbnail: ''
      },
      name: []
    }
  },
  userRole: {
    user: 'user',
    admin: 'admin',
    superadmin: 'superadmin'
  },
  userRoles: [
    { role: 'user', label: 'Юзер' },
    { role: 'admin', label: 'Адмін' },
    { role: 'superadmin', label: 'Суперадмін' }
  ],
  allowedforRegistrationRoles: ['admin', 'superadmin'],
  tabNames: {
    users: ['Користувачі', 'Адміністратори']
  },
  tableSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'medium'
  },
  iconSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'default'
  },
  colorCircleSizes: {
    SMALL_CIRCLE: '25px',
    DEFAULT_CIRCLE: '50px'
  },
  theme: {
    DARK_THEME: 'dark',
    LIGHT_THEME: 'light'
  },
  forbiddenRolesFromDeleting: ['superadmin'],
  dialogFormTitles: {
    REGISTER_ADMIN: 'Створити спецкористувача'
  },
  paginationPayload: {
    skip: 0,
    limit: 5,
    countPerPage: 6
  },
  sizesAvailableVariants: {
    AVAILABLE_TEXT: 'Доступний',
    UNAVAILABLE_TEXT: 'Відсутній'
  },
  pocketsAvailableVariants: {
    AVAILABLE_TEXT: 'Доступний',
    UNAVAILABLE_TEXT: 'Недоступний'
  },

  IMG_URL: 'https://horondi.blob.core.windows.net/horondi/images/',
  newsPerPage: 6,
  product: {
    sortBySelectOptions: [
      {
        label: 'Популярністю',
        value: 'popularity'
      },
      {
        label: 'Від дорогих до дешевих',
        value: 'sortDesc'
      },
      {
        label: 'Від дешевих до дорогих',
        value: 'sortAsc'
      },
      {
        label: 'Рейтингом',
        value: 'rate'
      }
    ],
    responsive: {
      superLargeDesktop: {
        breakpoint: { max: 4000, min: 3000 },
        items: 1
      },
      desktop: {
        breakpoint: { max: 3000, min: 1024 },
        items: 1
      },
      tablet: {
        breakpoint: { max: 1146, min: 464 },
        items: 1
      },
      mobile: {
        breakpoint: { max: 810, min: 0 },
        items: 1
      }
    }
  },
  popularity: 'popularity',
  rate: 'rate',
  sortAsc: 'sortAsc',
  sortDesc: 'sortDesc',
  submitKey: 'Enter',
  imagePrefix: 'https://horondi.blob.core.windows.net/horondi/images/',
  initialLanguageValues: [
    {
      lang: 'ua',
      value: ''
    },
    {
      lang: 'en',
      value: ''
    }
  ],
  doughnut: {
    colors: [
      colors.green[500],
      colors.red[700],
      colors.amber[600],
      colors.indigo[500]
    ],
    initialValues: {
      names: [],
      counts: [],
      relations: [],
      total: null
    }
  },
  bar: {
    initialValues: {
      labels: [],
      counts: []
    }
  },
  UAH: 'грн',
  maxItemsPerPage: 10,
  buttonTitles,
  detailTitles,
  messages,
  menuCategories,
  clientMenuCategories,
  catalogMenuCategories,
  constructorMenuCategories,
  staticPagesCategories,
  routes,
  statuses,
  tableHeadRowTitles,
  formRegExp,
  labels,
  titles,
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  categoryErrorMessages,
  patternErrorMessages,
  backErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  errorMessages,
  newsPaginationPayload,
  contactsPaginationPayload,
  modelErrorMessages,
  headerErrorMessages,
  statsErrorMessages,
  homePageSlideErrorMessages,
  constructorErrorMessages,
  paginationInputErrorMessages,
  commentErrorMessages,
  replyCommentErrorMessages,
  filterLabels,
  materialUiConstants,
  sizeErrorMessages,
  businessPageErrorMessages,
  imageUploadNewsInputsId,
  valueKeys,
  inputNames,
  pocketsErrorMessages
};
export const inputTypes = {
  button: 'button',
  submit: 'submit'
};

export const badgePosition = {
  vertical: 'top',
  horizontal: 'left'
};

export const dateFormat = ' HH:mm DD/MM/YYYY ';
export const dateFormatOrder = ' DD.MM.YYYY ';

export const formConstants = {
  formTypeSelect: 'select',
  isRequired: '*',
  notRequired: '',
  textFieldFilled: 'filled'
};
