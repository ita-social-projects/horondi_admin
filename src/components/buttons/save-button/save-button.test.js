import React from 'react';
import { mount } from 'enzyme';
import { noop } from 'lodash';
import SaveButton, { saveButtonHandler } from './save-button';
import { config, inputTypes } from '../../../configs';

const { SAVE_TITLE } = config.buttonTitles;

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: (selector) => selector(),
  useDispatch: () => jest.fn()
}));

describe('save button tests', () => {
  const size = 'small';
  const title = SAVE_TITLE;
  const type = inputTypes.button;
  const mockCallBack = jest.fn(noop());
  const mockUnblock = jest.fn();

  let component;

  beforeEach(() => {
    component = mount(
      <SaveButton
        size={size}
        title={title}
        type={type}
        onClickHandler={mockCallBack}
        unblockFunction={mockUnblock}
      />
    );
  });

  afterEach(() => {
    component.unmount();
  });

  it('should click', () => {
    expect(mockCallBack.mock.calls.length).toBe(0);
    component.find('button').simulate('click');
  });
  it('should exist and have value', () => {
    expect(component.prop('onClickHandler')).toEqual(mockCallBack);
  });
  it('should have type', () => {
    expect(component.find('button').type()).toEqual('button');
  });
  it('should have props', () => {
    expect(component.props().type).toBeDefined();
    expect(component.props().type).toEqual(type);
    expect(component.props().onClickHandler).toBeDefined();
    expect(component.props().onClickHandler).toBeTruthy();
  });
  it('should have propTypes', () => {
    expect(SaveButton.propTypes.title).toBeDefined();
    expect(SaveButton.propTypes.type).toBeDefined();
    expect(SaveButton.propTypes.onClickHandler).toBeDefined();
    expect(SaveButton.propTypes.unblockFunction).toBeDefined();
  });
  it('should have default props', () => {
    expect(SaveButton.defaultProps.onClickHandler).toEqual(noop);
  });
  it('should call saveButtonHandler', () => {
    const unblockFunctionMock = jest.fn();

    const result = saveButtonHandler(
      { unblockFunction: unblockFunctionMock },
      () => null,
      () => null,
      () => null,
      '',
      ''
    );
    result();

    expect(unblockFunctionMock).toHaveBeenCalled();
  });
});
