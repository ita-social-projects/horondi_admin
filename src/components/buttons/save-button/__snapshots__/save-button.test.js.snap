// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`save button tests should click 1`] = `
<SaveButton
  color="primary"
  errors={Object {}}
  onClickHandler={[MockFunction]}
  size="small"
  title="Зберегти"
  type="button"
  unblockFunction={[MockFunction]}
  values={Object {}}
>
  <WithStyles(ForwardRef(Button))
    color="primary"
    disabled={false}
    onClick={[Function]}
    size="small"
    type="button"
    unblockFunction={[MockFunction]}
    variant="contained"
  >
    <ForwardRef(Button)
      classes={
        Object {
          "colorInherit": "MuiButton-colorInherit",
          "contained": "MuiButton-contained",
          "containedPrimary": "MuiButton-containedPrimary",
          "containedSecondary": "MuiButton-containedSecondary",
          "containedSizeLarge": "MuiButton-containedSizeLarge",
          "containedSizeSmall": "MuiButton-containedSizeSmall",
          "disableElevation": "MuiButton-disableElevation",
          "disabled": "Mui-disabled",
          "endIcon": "MuiButton-endIcon",
          "focusVisible": "Mui-focusVisible",
          "fullWidth": "MuiButton-fullWidth",
          "iconSizeLarge": "MuiButton-iconSizeLarge",
          "iconSizeMedium": "MuiButton-iconSizeMedium",
          "iconSizeSmall": "MuiButton-iconSizeSmall",
          "label": "MuiButton-label",
          "outlined": "MuiButton-outlined",
          "outlinedPrimary": "MuiButton-outlinedPrimary",
          "outlinedSecondary": "MuiButton-outlinedSecondary",
          "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
          "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
          "root": "MuiButton-root",
          "sizeLarge": "MuiButton-sizeLarge",
          "sizeSmall": "MuiButton-sizeSmall",
          "startIcon": "MuiButton-startIcon",
          "text": "MuiButton-text",
          "textPrimary": "MuiButton-textPrimary",
          "textSecondary": "MuiButton-textSecondary",
          "textSizeLarge": "MuiButton-textSizeLarge",
          "textSizeSmall": "MuiButton-textSizeSmall",
        }
      }
      color="primary"
      disabled={false}
      onClick={[Function]}
      size="small"
      type="button"
      unblockFunction={[MockFunction]}
      variant="contained"
    >
      <WithStyles(ForwardRef(ButtonBase))
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        component="button"
        disabled={false}
        focusRipple={true}
        focusVisibleClassName="Mui-focusVisible"
        onClick={[Function]}
        type="button"
        unblockFunction={[MockFunction]}
      >
        <ForwardRef(ButtonBase)
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
          classes={
            Object {
              "disabled": "Mui-disabled",
              "focusVisible": "Mui-focusVisible",
              "root": "MuiButtonBase-root",
            }
          }
          component="button"
          disabled={false}
          focusRipple={true}
          focusVisibleClassName="Mui-focusVisible"
          onClick={[Function]}
          type="button"
          unblockFunction={[MockFunction]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
            unblockFunction={[MockFunction]}
          >
            <span
              className="MuiButton-label"
            >
              Зберегти
            </span>
            <WithStyles(memo)
              center={false}
            >
              <ForwardRef(TouchRipple)
                center={false}
                classes={
                  Object {
                    "child": "MuiTouchRipple-child",
                    "childLeaving": "MuiTouchRipple-childLeaving",
                    "childPulsate": "MuiTouchRipple-childPulsate",
                    "ripple": "MuiTouchRipple-ripple",
                    "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                    "rippleVisible": "MuiTouchRipple-rippleVisible",
                    "root": "MuiTouchRipple-root",
                  }
                }
              >
                <span
                  className="MuiTouchRipple-root"
                >
                  <TransitionGroup
                    childFactory={[Function]}
                    component={null}
                    exit={true}
                  />
                </span>
              </ForwardRef(TouchRipple)>
            </WithStyles(memo)>
          </button>
        </ForwardRef(ButtonBase)>
      </WithStyles(ForwardRef(ButtonBase))>
    </ForwardRef(Button)>
  </WithStyles(ForwardRef(Button))>
</SaveButton>
`;
